core/embed/rust/src/ui/model_tr/component/frame.rs

```diff
@@ -66,8 +66,14 @@ where
     fn place(&mut self, bounds: Rect) -> Rect {
         const TITLE_SPACE: i16 = 2;
 
-        let (title_area, content_area) = bounds.split_top(theme::FONT_HEADER.line_height());
-        let content_area = content_area.inset(Insets::top(TITLE_SPACE));
+        // Title might be empty, in that case allocating whole area for the content
+        let (title_area, content_area) = if self.title.get_text().is_empty() {
```

Comment:
could we instead convert title to `Option`?

Reply:
Theoretically yes, but I would rather leave it like this - it is just one corner-case when the title is empty.

I would say it is not worth complicating the code because of it.

Reply:
but this way there is hidden behavior (that you need to explicitly sidestep elsewhere)

given the "centered" option, i'm thinking maybe the "show passphrase" screen should be its own thing instead of hacking "show_info" to support it

Reply:
Right, will go with new `show_passphrase` for now. After merging all the  new`T2B1` designs, we might want to take a closer look at the API and maybe unify it more with `TT` - we use `show_simple` in that case (but also getting optional title).

Reply:
Done in [48bf66b](https://github.com/trezor/trezor-firmware/pull/3098/commits/48bf66b51097fc92c1fcfc2963c4552a9a236250)